Source/AwsCommonRuntimeKit/Utilities.swift

```diff
@@ -41,6 +41,21 @@ extension String {
         let byteCursor = aws_byte_cursor_from_buf(output)
         return byteCursor.toData().base64EncodedString()
     }
+
+    public func base64EncodedSha256(allocator: Allocator = defaultAllocator, truncate: Int = 0) -> String? {
```

Comment:
Could this just return `String` instead of `String?`
I don't think it can ever actually fail. aws_sha256_compute() can only fail if allocation fails (that's fatal now) or the output buffer isn't big enough (you're creating the output buffer, so there's no chance of messing that up)

I understand if you want to stick with `String?` because the other existing `base64EncodedMD5()` function returns `String?`